Kurdish translations

roj1512 opened the pull request:
WIP

GioSensation commented:
Hey @roj1512. Thanks for your contribution. Yes, this is the right way to add the locale. You will also need to add entries to the readme and the locales.json list. See this PR as an example: https://github.com/duckduckgo/duckduckgo-locales/pull/535/files.

Before proceeding, though, I will ask you to check whether our search engine supports queries in your language. Try performing searches in your language and script and make sure we are returning useful results. From what I could see from a couple of tests, we do, but I'd like your perspective as a native speaker.

I'll be out until next week, but I will check back again then.

Thanks again for looking into this. It's really appreciated.

roj1512 commented:
Hi @GioSensation, thank you so much for your confirmation!

> From what I could see from a couple of tests, we do, but I'd like your perspective as a native speaker.

Of course, that's the main reason we're interested in having it in Kurdish.

> You will also need to add entries to the readme and the locales.json list.

Will do :)

roj1512 commented:
@GioSensation Will it be possible to provide us an endpoint to preview our changes?

CYAXXX commented:
> @GioSensation Will it be possible to provide us an endpoint so we can preview our translations?

@GioSensation  We still waiting for an answer, we stopped the translating progress in result we can figure if we are doing the right thing, and at the same time we were seeking for a preview and test maybe if that works

GioSensation commented:
> @GioSensation Will it be possible to provide us an endpoint so we can preview our translations?

We can't provide an endpoint, but I can take screenshots for you to evaluate. I will get back to this on Wednesday. Thank you for your patience.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5938 @@
+"permanent way (on a remote server in the cloud). No personally identifiable "
+"information will be stored in the cloud, and your passphrase will never "
+"leave your browser."
+msgstr "Bi berdestî, sazkarî wê werin tomarkirin li ser kilorên gerokê yên ne kesane (di "gerokê" te de). 
```
Please do not use double quotes within the strings. It will break the format and make the file unusable. I suggest you use a specified .po editor like https://poedit.net/ (there is a free version). It will prevent such problems.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5938 @@
+msgid ""
+"Browse as usual, and we'll take care of the rest. We bundled our search "
+"engine, tracker blocker, and encryption enforcer into one %s%s Extension%s."
+msgstr "Geranê xwe wekî her dem bike, û em ê yên mayî li ser me ye. Me motora lêgerînê xwe, astengkera şopgeran, û şîfrekirin di yek %s%sPêvek de pêk anî."
```
The source string has 3 placeholders (the `%s` symbol), but the translated string only has 2. This breaks our translation system, so please make sure to always match the number of placeholders. According to our tests there are other such instances, so please check the file using a specialised software like [Poedit](https://poedit.net/). It will highlight such inconsistencies for you. 🙂

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5938 @@
+
+msgctxt "Ads GDPR, internal use only. Do not change"
+msgid "More Info"
+msgstr "Bêtir Zanyarî"
```
Please keep this string untranslated. It's for internal use only. Thanks! 🙂
```suggestion
msgstr ""
```

roj1512 replied:
Fixed!

GioSensation reviewed (changes requested):
I gave this a first pass. There are a few issues with the translations. Do you mind taking a look? Do not hesitate to post a comment if something is unclear. I'll be happy to help. 🙂

roj1512 commented:
@GioSensation Please lemme know if the problems are fixed and we're ready for a preview :)

CYAXXX commented:
Any updates?

roj1512 commented:
@CYAXXX That guy is busy.

CYAXXX commented:
> @CYAXXX That guy is busy.

Absolutely man, all what we need is yes or no it's been months which we are waiting for DDG

GioSensation commented:
Hey there, yeah, pretty busy 😅. I see the tests are still failing. I will now comment directly on all the strings, but again a .po editor would help catch most of these problems. Also, as mentioned previously, you should also add the new locale to the README and the locale.json file as in this PR https://github.com/duckduckgo/duckduckgo-locales/pull/535/files.

CYAXXX commented:
> Hey there, yeah, pretty busy 😅. Thanks for pinging me again. I see the tests are still failing. I will now comment directly on all the strings, but again a .po editor would help catch most of these problems. Also, as mentioned previously, you should also add the new locale to the README and the locale.json file as in this PR https://github.com/duckduckgo/duckduckgo-locales/pull/535/files.

@roj1512 can you please take care of exist problems cause I didn't figure it.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+msgstr "%s li ser %s"
+
+msgctxt "install-duckduckgo"
+msgid "%sClick %sAdd%sCheck %sAllow%s, then click %sOkay%s"
+msgstr "%sMafê bide %sKontrola%sTevlîkirinê%s, paşê %sBaş e%s Bitikîne"
+
+msgctxt "install-duckduckgo"
+msgid "%sClick %sAllow%sClick %sAdd%sCheck %sAllow%s, then click %sOkay%s"
+msgstr "%sMafê bide %sKontrola%sTevlîkirinê%s, paşê %sBaş e%s Bitikîne"
```
In both of these strings the translations don't contain the same number of placeholders than the source. It seems that the two translations are the same, but the source strings are different.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+msgid ""
+"%sClick %sContinue To Installation%sClick %sAdd%sCheck %sAllow%s, then click "
+"%sOkay%s"
+msgstr "%sBidomîne %sBitikîne da ku Saz bikî %sMafê bide %sKontrola%sTevlîkirinê, paşê %sBaş e%s bitikîne"
```
Same here. The placeholders don't match.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+msgstr "%smezinker%s bitikîne di darika lêgerînê de"
+
+msgid "Click on the magnifying glass in the search box at the top right"
+msgstr "Mezinker%s bitikîne di kodika lêgerînê de li jor rastê"
```
Same here, there's a placeholder after `Mezinker` that shouldn't be there.

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+msgid ""
+"Click or tap %sDelete My Data%s. This removes the data from the cloud, but "
+"it remains in your browser until you click/tap %sReset All Settings%s."
+msgstr "% Daneyên Min jê bibe%s bitikîne an jî pêl bike. Ev daneyên ji ewrê jê dibe, lê "
```
```suggestion
msgstr "s%Daneyên Min jê bibe%s bitikîne an jî pêl bike. Ev daneyên ji ewrê jê dibe, lê "
```

GioSensation commented:
> @roj1512 can you please take care of exist problems cause I didn't figure it.

I can't really, because I don't know the language. But I've posted specific comments where the issues are happening. The translation strings should have the same number of placeholders (`%s`) of the source string.

CYAXXX commented:
> > @roj1512 can you please take care of exist problems cause I didn't figure it.
> 
> I can't really, because I don't know the language. But I've posted specific comments where the issues are happening. The translation strings should have the same number of placeholders (`%s`) of the source string.



> > @roj1512 can you please take care of exist problems cause I didn't figure it.
> 
> I can't really, because I don't know the language. But I've posted specific comments where the issues are happening. The translation strings should have the same number of placeholders (`%s`) of the source string.

No no it's okey, I was asking @roj1512 cause we both are working on this

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+msgstr "%sSembola alavan%s bitikîne di jor rastê ji dîmendera gerokê"
+
+msgid "Click on the %smagnifying glass%s in the search bar"
+msgstr "%smezinker bitikîne di darika lêgerînê de"
```
Now this one is missing one placeholder 🙂.

CYAXXX replied:
@roj1512 will provide missing data soon

GioSensation commented on locales/ku_KU/duckduckgo.po:
```diff
@@ -0,0 +1,5940 @@
+"permanent way (on a remote server in the cloud). No personally identifiable "
+"information will be stored in the cloud, and your passphrase will never "
+"leave your browser."
+msgstr "Bi berdestî, sazkarî wê werin tomarkirin li ser kilorên gerokê yên ne kesane (di "
+"gerokê te de). Tu dikarî Tomarkirina Ewrî ya nenas bi kar bînî "
+"riya mayînde (li ser rajekareke dûr di ewrê de).
+Ne kesayeteke naskirî ye"
+"Zanyarî wê di ewrê de werin tomarkirin û hevoka te wê ti carî nehêle " 
+"geroka xwe bihêle."
```
```suggestion
msgstr "Bi berdestî, sazkarî wê werin tomarkirin li ser kilorên gerokê yên ne kesane (di "
"gerokê te de). Tu dikarî Tomarkirina Ewrî ya nenas bi kar bînî "
"riya mayînde (li ser rajekareke dûr di ewrê de). Ne kesayeteke naskirî ye"
"Zanyarî wê di ewrê de werin tomarkirin û hevoka te wê ti carî nehêle " 
"geroka xwe bihêle."
```

CYAXXX commented:
@GioSensation please can you check for provided data? And if still there's any problem..

roj1512 commented:
@CYAXXX Yes, there is. The CI is failing.

CYAXXX commented:
> @CYAXXX Yes, there is. The CI is failing.

Çare çi ye?

roj1512 commented:
@GioSensation The CI is passing now 🎉☺️

GioSensation commented:
Thanks guys. I will integrate the translations in a development environment and send you a bunch of screenshot so you can confirm everything is looking fine. I can't get to this today, but I'll set a reminder for this Friday. Thanks for your patience.

GioSensation commented:
Hello people! So, I've installed the language in a development environment and the translations are looking pretty good from what I can see 🎉👏. Here is a screenshot of our search result page.

![CleanShot 2022-10-14 at 16 46 32](https://user-images.githubusercontent.com/1828326/195875790-2983c751-f92e-49e8-9ebc-fb907d131bff.jpg)

I need you to please move the file into a subfolder the same way the other language files are.

Current path:

```
locales/ku_KU/duckduckgo.po
```

Correct path:

```
locales/ku_KU/LC_MESSAGES/duckduckgo.po
```

Other than that, we have two ways of moving forward. Either we merge the current PR and you guys open new PRs with more translations, or we keep this PR open and you guys add translations right here in this PR. It really is up to you. Let me know and please make that path change mentioned above. Thank you for your contribution and again thanks for your patience.

roj1512 commented:
@GioSensation If we merge this, will it be live?

roj1512 commented:
@GioSensation OK, we can go with merging this.

GioSensation commented:
I will merge this on Wednesday. Please add as many strings as possible before then, so we can provide a better UX for our users when this is released. I noticed we're missing several strings even on the home and search result page. Thanks!

GioSensation commented:
Hello, folks. So, I've pushed a few commits to this branch. Mainly:

- I have removed strings that were not translated but copied over from English (see the [commit](https://github.com/duckduckgo/duckduckgo-locales/pull/554/commits/bc06cfadec1924c9cdc669cc1d52801ca6ad397f)). In the future, please leave empty strings instead of copying over the English version. This enables future translators to use .po editors to see exactly what strings are missing. Do not worry, our system falls back to English if translations are missing so nothing changes for the end users.
- I have renamed `ku_KU` to `ku_TR` which seems to be the accepted version of the language code.

If you're happy with the changes, thumb this message up and I will merge. Otherwise do let me know.

Thank you again for your contribution and for your patience while working with me in getting this PR ready. 💪❤️

roj1512 commented:
Hi, @GioSensation

> I have removed strings that were not translated but copied over from English (see the [commit](https://github.com/duckduckgo/duckduckgo-locales/pull/554/commits/bc06cfadec1924c9cdc669cc1d52801ca6ad397f)). In the future, please leave empty strings instead of copying over the English version. This enables future translators to use .po editors to see exactly what strings are missing. Do not worry, our system falls back to English if translations are missing so nothing changes for the end users.

Thanks!

> I have renamed `ku_KU` to `ku_TR` which seems to be the accepted version of the language code.

We’re happy that you noticed and corrected the wrong country code we intentionally used. If the country code is mandatory, we prefer `SY` or `IQ`, like `ku_SY` or `ku_IQ`, since we, the translators are from the Kurdistani parts of those countries. It’s also totally normal if you’re not okay with it, it’s your project at the end of the day.

> If you're happy with the changes, thumb this message up and I will merge. Otherwise do let me know.

We’re happy, just keep the note above in mind :)

Here’s the thumbs up for you: 👍 

> Thank you again for your contribution and for your patience while working with me in getting this PR ready. :muscleheart:

You’re welcome af!

GioSensation commented:
Thanks. I have removed the country code. We can evaluate options if there's any problem but from tests on the dev environment things are looking good. I am merging the PR. You should see the option to select the Kurdish language tomorrow. Please ping me if anything seems wrong and I urge you to please continue to add strings so that speakers of your language can have a better experience on our search engine. Thank you!

roj1512 commented:
> I have removed the country code.

Better!
